sdk/servicebus/azure-messaging-servicebus/src/samples/java/com/azure/messaging/servicebus/ReceiveMessageAndSettleSample.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.messaging.servicebus.models.ReceiveMode;
+import reactor.core.Disposable;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Sample demonstrates how to receive an {@link ServiceBusReceivedMessage} from an Azure Service Bus Queue and settle
+ * it. Settling of message include accept, defer and abandon the message as needed.
+ */
+public class ReceiveMessageAndSettleSample {
+    private static final Duration TIME_OUT = Duration.ofSeconds(15);
+
+    /**
+     * Main method to invoke this demo on how to receive an {@link ServiceBusMessage} from an Azure Service Bus
+     * Queue
+     *
+     * @param args Unused arguments to the program.
+     */
+    public static void main(String[] args) {
+
+        // The connection string value can be obtained by:
+        // 1. Going to your Service Bus namespace in Azure Portal.
+        // 2. Go to "Shared access policies"
+        // 3. Copy the connection string for the "RootManageSharedAccessKey" policy.
+        String connectionString = System.getenv("AZURE_SERVICEBUS_CONNECTION_STRING");
+
+        // Create a receiver.
+        // "<<fully-qualified-namespace>>" will look similar to "{your-namespace}.servicebus.windows.net"
+        // "<<queue-name>>" will be the name of the Service Bus queue instance you created
+        // inside the Service Bus namespace.
+
+        ServiceBusReceiverAsyncClient receiverAsyncClient = new ServiceBusClientBuilder()
+            .connectionString(connectionString)
+            .buildReceiverClientBuilder()
+            .receiveMode(ReceiveMode.PEEK_LOCK)
+            .isLockAutoRenewed(true)
+            .queueName("<<queue-name>>")
+            .isAutoComplete(false)
+            .maxAutoLockRenewalDuration(Duration.ofSeconds(2))
+            .buildAsyncClient();
+
+        Disposable subscription = receiverAsyncClient.receive()
+            .subscribe(received -> {
```

Comment:
I don't think this is a good sample, I think it would be better to show chained reactor operations that could take longer than the 15 seconds.

For example:
```java
client.receive().flatMap(message -> {
    // A sample of processing the message followed by a persisting the message logic.
    return Mono.delay(Duration.ofSeconds(15).thenReturn(message);
}).onErrorContinue((message, error) -> {
    // Handle the message that caused the error while processing above. In this case, we'll abandon it and let someone else handle it.
    client.abandon(message).block();
})
.flatMap(message -> {
  // This is a message that was successfully processed. So we complete it.
  return client.complete(message);
})
.subscribe();
```

Reply:
I do not think we can get message in `onErrorContinue`  . But I have changed the sample as you told.